lightly/utils/io.py

```diff
@@ -91,6 +94,18 @@ def check_embeddings(path: str):
                     f'Found empty row on line {i}.'
                     )
 
+    if remove_additional_columns:
+        new_rows = []
+        with open(path, 'r', newline='') as csv_file:
```

Comment:
Why do you reopen the same file? Can you not just continue reading the already opened file and add the header to new_rows?

Reply:
No, I cannot, as the file was already fully read by iterating over all rows.

Reply:
Oh wow, bad review from my side. You could, however, make the row length check at the same time as creating the `new_rows` list to avoid reading twice.